mllib/src/main/scala/org/apache/spark/ml/regression/LinearRegression.scala

```diff
@@ -212,12 +230,139 @@ class LinearRegressionModel private[ml] (
   extends RegressionModel[Vector, LinearRegressionModel]
   with LinearRegressionParams {
 
+  @transient private var trainingSummary: Option[LinearRegressionTrainingSummary] = None
```

Comment:
What about during model save/load? Users may want to see some training stats such as r^2 and MSE after loading a model. If `trainingSummary` is transient and the only place we put the small stats, we will need to think of another way to support this use case.